content/blog/2019/egress-traffic-control-in-istio-part-2/index.md

```diff
@@ -0,0 +1,116 @@
+the [Kubernetes network policies](https://kubernetes.io/docs/concepts/services-networking/network-policies/) or an L3
+firewall that will enforce that traffic from the cluster to the outside is allowed for the egress gateway only. See
+an example of the [Kubernetes network policies configuration](/docs/tasks/traffic-management/egress/egress-gateway/#apply-kubernetes-network-policies).
+
+You must also increase the security measures applied to the Istio control plane pods and the egress gateway, for example:
+
+* run them on nodes separate from the application nodes
+* run them in a separate namespace
+* apply Kubernetes RBAC and Network policies to protect them
+* monitor them more thoroughly and more frequently than the application pods
+
+After all, if the attackers are able to attack Istio Mixer or the egress gateway, they could
+violate any policy.
+
+Once you direct egress traffic through an egress gateway and apply the additional security mechanisms,
+you can securely monitor and enforce security policies for the traffic.
+
+The following diagram shows Istio's security architecture, augmented with an L3 firewall which is part of the
+[additional security mechanisms](/docs/tasks/traffic-management/egress/egress-gateway/#additional-security-considerations)
+that should be provided outside of Istio.
+
+{{< image width="80%" link="./SecurityArchitectureWithL3Firewalls.svg" caption="Istio Security Architecture with Egress Gateway and L3 Firewall" >}}
+
+You can configure the L3 firewall trivially to only allow incoming traffic through the Istio ingress gateway and
+only allow outgoing traffic through the Istio egress gateway. The Istio proxies of the gateways enforce policies and report telemetry as all other proxies in the mesh.
+
+Now let's examine possible attacks and let me show you how the secure control of egress traffic in Istio prevents them.
+
+## Preventing possible attacks
+
+Consider the following security policies for egress traffic:
+
+* Application **A** is allowed to access `*.ibm.com`, which includes all the external services with URLs matching
+`*.ibm.com`.
+* Application **B** is allowed to access `mongo1.composedb.com`.
+* All egress traffic is monitored.
+
+Suppose the attackers have the following goals:
+
+1. To access `*.ibm.com` from your cluster.
+1. To access `*.ibm.com` from your cluster, unmonitored. The attackers want their traffic to be unmonitored to prevent a
+possibility that you will detect the forbidden access.
+1. To access `mongo1.composedb.com` from your cluster.
+
+Now suppose that the attackers manage to break into one of the pods of application **A**. The attackers try to use the compromised pod to perform the forbidden access.
+
+First, let's see which attacker's goals it is possible for you to thwart:
+
+1. Since application **A** is allowed to access `*.ibm.com`, the compromised pod is able to access it. There is no way to prevent such access since there is no way to distinguish, at
```

Comment:
@vadimeisenbergibm I'm not sure these three items should be a list at all, let a lone an ordered list since they are not things that happen in sequence or that readers need to follow in a specific order. 
IMO, it could work as either a bullet list or just a paragraph. I favor the former.
The way the list is ordered, gives readers the impression that all you described was for nothing. My suggestion reorders the first item on the list to avoid that.
 ```suggestion
- Initially, there is no way to prevent a compromised application **A** to access `*.ibm.com`, because the compromised pod is able to access it indistinguishable from the original pod.
```

Reply:
@rcaballeromx Let me provide a high level flow of this subsection:

1. You have three security policies regarding the egress traffic in an example organization
2. The attackers manage to break into a pod (compromise it) and try to access external services
3. The attackers have three goals (three is accidental, not related to the number of policies).
4. You cannot prevent the first goal initially, only after you monitor and detect unusual traffic. You can prevent the second and third goals from the start.
5. The attackers have three possible attacks (that I know). Here three is accidental, the number of attacks does not have to correspond to the number of policies and the number of goals.

Reply:
@rcaballeromx this is why I wanted the list of goals and the list of what can be prevented as the numbered list, since they are related.